support-lambdas/bigquery-acquisitions-publisher/typescript/acquisitions.ts

```diff
@@ -0,0 +1,137 @@
+import { z } from 'zod';
+import {
+	ProductTypeSchema,
+	IsoCountrySchema,
+	IsoCurrencySchema,
+	ContributionTypeSchema,
+	PaymentMethodSchema,
+	IsoCountryType,
+	ContributionType,
+	ProductType,
+	IsoCurrencyType,
+	PaymentMethodType,
+} from './dependencies';
+
+// Items to write to BigQuery (datalake:fact_acquisition_event)
+export const AcquisitionProductSchema = z.object({
+	eventTimeStamp: z.string(),
+	country: z.enum(IsoCountrySchema),
+	componentId: z.string().nullable(),
+	componentType: z.string().nullable(),
+	campaignCode: z.string().nullable(),
+	referrerUrl: z.string().nullable(),
+	abTests: z.object({ name: z.string(), variant: z.string() }).array(),
+	paymentFrequency: z.enum(ContributionTypeSchema),
+	paymentProvider: z.enum(PaymentMethodSchema),
+	printOptions: z
+		.object({ product: z.string(), delivery_country_code: z.string() })
```

Comment:
I think the input here will be `deliveryCountryCode` not `delivery_country_code`.

Reply:
Also I think it's worth typing product to match the `PrintProduct` class here:
https://github.com/guardian/support-frontend/blob/906b5622475c1d1fc10ba42177a383836a16659f/support-modules/acquisition-events/src/main/scala/com/gu/support/acquisitions/models/AcquisitionDataRow.scala#L144